src/screens/SendScreen.tsx

```diff
@@ -68,113 +81,119 @@ const SendScreen = ({
   const theme = useTheme()
   const mainAddress = useAppSelector((state) => state.addresses.mainAddress)
-  const [fromAddressHash, setFromAddressHash] = useState<AddressHash>(addressHash ?? mainAddress)
-  const fromAddress = useAppSelector((state) => selectAddressByHash(state, fromAddressHash))
-  const [toAddressHash, setToAddressHash] = useState<string>('')
-  const [amountString, setAmountString] = useState('')
   const [amount, setAmount] = useState(BigInt(0))
   const [fees, setFees] = useState<bigint>(BigInt(0))
   const [unsignedTxId, setUnsignedTxId] = useState('')
   const [unsignedTransaction, setUnsignedTransaction] = useState('')
-  const [gasAmount, setGasAmount] = useState('')
-  const [gasPriceString, setGasPriceString] = useState('')
-  const [gasPrice, setGasPrice] = useState<bigint>()
   const [isLoading, setIsLoading] = useState(false)
   const [isConsolidateUTXOsModalVisible, setIsConsolidateUTXOsModalVisible] = useState(false)
   const [consolidationRequired, setConsolidationRequired] = useState(false)
   const [isSweeping, setIsSweeping] = useState(false)
   const [sweepUnsignedTxs, setSweepUnsignedTxs] = useState<SweepAddressTransaction[]>([])
   const dispatch = useAppDispatch()
   const [txStep, setTxStep] = useState<TxStep>('build')
-  const totalAmount = amount + fees
-
-  const isFormDataComplete = amount > BigInt(0) && isAddressValid(toAddressHash) && isAddressValid(fromAddressHash)
-
-  const gasAmountHasError = !!gasAmount && parseInt(gasAmount) < MINIMAL_GAS_AMOUNT
-  const gasPriceHasError = !!gasPrice && gasPrice < MINIMAL_GAS_PRICE
-
-  const handleAmountChange = (str: string) => isNumericStringValid(str) && setAmountString(str)
-  const handleGasAmountChange = (str: string) => isNumericStringValid(str, false) && setGasAmount(str)
-  const handleGasPriceChange = (str: string) => isNumericStringValid(str) && setGasPriceString(str)
-
-  useEffect(() => {
-    try {
-      setAmount(convertAlphToSet(amountString))
-    } catch {
-      setAmount(BigInt(0))
+  const {
+    control,
+    watch,
+    handleSubmit,
+    formState: { errors }
+  } = useForm<FormData>({
+    defaultValues: {
+      fromAddressHash: addressHash ?? mainAddress,
+      toAddressHash: '',
+      amountInAlph: '',
+      gasAmount: '',
+      gasPriceInAlph: ''
     }
-  }, [amountString])
+  })
+  const { fromAddressHash, toAddressHash, gasAmount, amountInAlph, gasPriceInAlph } = watch()
+  const fromAddress = useAppSelector((state) => selectAddressByHash(state, fromAddressHash))
 
-  useEffect(() => {
-    try {
-      setGasPrice(convertAlphToSet(gasPriceString))
-    } catch {
-      setGasPrice(undefined)
-    }
-  }, [gasPriceString])
+  const isFormValid = isEmpty(errors)
+  const totalAmount = amount + fees
+
+  const validateIsAddressValid = (value: string) => isAddressValid(value) || 'This address is not valid'
+  const validateIsNumericStringValid = (value: string) => isNumericStringValid(value) || 'We need a number here!'
```

Comment:
Should we start to extract generic validation method out of this component?

Also, let's maybe replace the error by something like `A number is expected.` :)

Reply:
done!